data/render/src/types.rs

```diff
@@ -62,7 +62,10 @@ pub struct Catalog {
 mod filters {
     // eventually, if other open-source sites (e.g. gitlab) support something like stars, those
     // could also be formatted in this filter
-    pub fn format_badge(s: &str) -> ::askama::Result<String> {
+    pub fn format_badge(mut s: &str) -> ::askama::Result<String> {
+        if s.chars().last().unwrap() == '/' {
```

Comment:
I think `?` instead of `.unwrap()` should work here, too and it would be a bit more idiomatic.
`unwrap` panics in case of an error (i.e. when `s` is an empty string) whereas `?` propagates the error up the chain where it can be handled (e.g. by an error message).

Reply:
I edit the code as shown below
```
if s.chars().last()? == '/' {
    s = s.trim_end_matches('/');
}
```
but I got this error
```
Compiling render v0.1.0 (/mnt/c/Users/jj251/Desktop/static-analysis/data/render)
error[E0277]: `?` couldn't convert the error to `askama_shared::error::Error`
  --> src/types.rs:66:28
   |
66 |         if s.chars().last()? == '/' {
   |                            ^ the trait `std::convert::From<std::option::NoneError>` is not implemented for `askama_shared::error::Error`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the following implementations were found:
             <askama_shared::error::Error as std::convert::From<std::fmt::Error>>
   = note: required by `std::convert::From::from`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `render`.

To learn more, run the command again with --verbose.
```
I google it a little bit, but still having no idea about how to fix this bug. If you have any suggestion, please let me know.

BTW, I found that in ```data/render/templates/README.md```, there is ```{% if linter.source.is_some() %}{{ linter.source.as_ref().unwrap()|format_badge }}{%endif%}```, I think that will prevent from empty string, so maybe that is ok to use ```unwrap()```?